src/pages/publication/ReferencesPanel.tsx

```diff
@@ -31,42 +31,44 @@ interface ReferencesPanelProps {
 const ReferencesPanel: React.FC<ReferencesPanelProps> = ({ goToNextTab, savePublication }) => {
   const { t } = useTranslation('publication');
   const { values, setFieldTouched, setFieldValue }: FormikProps<FormikPublication> = useFormikContext();
-  const { publicationType } = values.entityDescription;
-
-  // Validation messages won't show on fields that are not touched
-  const setAllFieldsTouched = useCallback(() => {
-    Object.values(ReferenceFieldNames).forEach((fieldName) => setFieldTouched(fieldName));
-  }, [setFieldTouched]);
+  const { publicationType, publicationSubtype } = values.entityDescription;
 
   useEffect(
-    () => () => {
-      // Set all fields as touched if user navigates away from this panel (on unmount)
-      setAllFieldsTouched();
-    },
-    [setAllFieldsTouched]
+    // Set all fields as touched if user navigates away from this panel (on unmount)
+    () => () => Object.values(ReferenceFieldNames).forEach((fieldName) => setFieldTouched(fieldName)),
+    [setFieldTouched]
   );
 
+  useEffect(() => {
+    // Update publicationInstance's type when publicationSubtype changes
+    setFieldValue(ReferenceFieldNames.PUBLICATION_INSTANCE_TYPE, publicationSubtype);
+  }, [setFieldValue, publicationSubtype]);
+
   return (
     <TabPanel ariaLabel="references" goToNextTab={goToNextTab} onClickSave={() => savePublication()}>
       <StyledSelectContainer>
         <SelectTypeField
           fieldName={ReferenceFieldNames.PUBLICATION_TYPE}
           options={Object.values(PublicationType)}
-          onChangeExtension={() => setFieldValue(ReferenceFieldNames.SUB_TYPE, '')}
+          onChangeExtension={() => {
```

Comment:
hva betyr egentlig onChangeExtension? hva er `extension`? 😅  kan kanskje renames til noe litt mer fornuftig?

Reply:
uff ja. Det er noe som skjer i tillegg til (etter) den vanlige onChange til en field (som setter verdi, touched, etc): https://github.com/BIBSYSDEV/NVA-Frontend/blob/48a4ff0b585cc0aca6446a67db4b05c5fc32b6bd/src/pages/publication/references_tab/components/SelectTypeField.tsx#L30

Prøvde å fjerne denne helt og flytte det til en useEffect, men det gikk heller ikke. Jeg kommer ikke på noe bedre navn, forslag?

Reply:
Hva med `onChangeSubtype`?

Reply:
Da syns jeg det virker som at det bare er det som skjer.
For å oppsummere `SelectTypeField` så brukes den både for å velge hovedtype og subtype. Når brukeren endrer hovedtype vil vi også nullstille valgt tidsskrift og subtype siden de (trolig) ikke er gyldig for nyvalgt hovedtype uansett.
Når brukeren velger subtype vil vi ikke gjøre noe ekstra

Reply:
hva med `onChangeType` da?

Reply:
syns fortsatt det høres ut som at den da overskriver hele onChange, men den er ikke verre enn `onChangeExtension` så jeg kan bli med på den! :D

Reply:
Eller, subtype er jo også en type da... så `onChangePublicationType`?